sw/device/lib/testing/rand_testutils.c

```diff
@@ -82,5 +83,24 @@ uint32_t rand_testutils_gen32_range(uint32_t min, uint32_t max) {
   if (range == 0) {
     return min;
   }
-  return min + (rand_testutils_gen32() % (range + 1));
+  uint32_t result = min + (rand_testutils_gen32() % (range + 1));
+  CHECK(result >= min && result <= max);
+  return result;
+}
+
+void rand_testutils_shuffle(void *array, size_t size, size_t length) {
+  if (length <= 1) {
+    return;
+  }
+  unsigned char temp[size];
+  unsigned char *array8 = array;
+  uint32_t reseed_frequency = rand_testutils_rng_ctx.reseed_frequency;
+  rand_testutils_rng_ctx.reseed_frequency = UINT32_MAX;
+  for (size_t i = length - 1; i > 0; i--) {
+    size_t j = rand_testutils_gen32_range(0, i + 1);
```

Comment:
Since the range used in `rand_testutils_gen32_range` is [inclusive](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/rand_testutils.h;drc=6d1e7bc337d1a075d8841568614653b9da3e000a;l=94), and `i` starts at `length - 1`, isn't there a chance `j` is outside the bounds of the source array (`array` or also `array8`)?